wunderground_upload.yaml

```diff
@@ -214,8 +214,12 @@ variables:
     {{ sensor_windgust if sensor_windgust == ''
         else (states(sensor_windgust) | float(default=0)) * (0.6213 if unit_of_measurement == 'km/h' else 1) }}
   winddir: >
-    {{ sensor_winddir if sensor_winddir == ''
-        else (states(sensor_winddir) | float(default=0)) }}
+    {% set candidate_wind_direction = states(sensor_winddir) %}
+    {% if not candidate_wind_direction is number %}
```

Comment:
The value is always a string, so this if-clause is always true. When an angle is supplied instead of an direction, there is no output.

To fix this, I would suggest changing this line to:
```YAML
    {% set wind_direction_float = candidate_wind_direction | float(default=0) %}
    {% if wind_direction_float == 0 %}
```

If I test this with an angle and a direction sensor, the output works in both cases.

Reply:
I updated as you suggested but used -1 instead of 0 as the marker of textual candidate_wind_direction, so that a 0 (North) isn't going into the text to angle mapping.
I'm just a bit worried that -1 could be returned by some winddir sensors... maybe I should have used something like -999?

Reply:
The value of -1 should be enough. 0 is an angle that could happen, -1 not since it will be 359.